### PR TITLE
feat(trainings): add printing format

### PR DESCRIPTION
**Pull request by jsfez**

### CODE REVIEW

File: src/client/TrainingPdf.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        </ProgrammeDayTitle>
+        <ReactMarkdown source={day} />
+      </ProgrammeDay>
+    </Planning>,
+  )
+
+  return (
+    <Planning>
+      {res}
+      {/* {<ReactMarkdown source={daySplitting[1]} />} */}
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: No comment please.

File: src/client/TrainingPdf.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    </Planning>
+  )
+}
+
+const Description = ({ descriptionContent }) => {
+  const descriptionSplitting = descriptionContent.split('### Objectifs')
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Not enough generic `/###.*\n/` would be better IMO.

File: src/client/TrainingPdf.js [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+  font-weight: bold;
+  border-left: 4px ${theme.colors.primary} solid;
+  padding-left: 10px;
+  font-size: 12px;
+  color: black;
+  line-height: 1em;
+`
+
+const ProgrammeArray = ({ programmeContent }) => {
+  const daySplitting = programmeContent.split(/### Jour \d+/)
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Not enough generic, same.

File: src/client/TrainingPdf.js [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+  li {
+    margin: 0;
+    line-height: 1.2em;
+  }
+`
+const ProgrammeDayTitle = styled(SectionTitle)`
+  margin: 0 0 5px 0;
+  padding: 0;
+  ${'' /* margin-bottom:5px; */}
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: No comment please.

File: src/client/TrainingPdf.js [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+  font-weight: bold;
+  border-left: 4px ${theme.colors.primary} solid;
+  padding-left: 10px;
+  font-size: 12px;
+  color: black;
+  line-height: 1em;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Just without unit it's fine.

File: src/client/TrainingPdf.js [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+  h3 + p {
+    margin: 0;
+  }
+`
+
+const Objectives = ({ description }) => {
+  const descriptionSplitting = description
+    .replace(/- /g, '')
+    .split('### Objectifs')
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Generic..

File: src/client/TrainingPdf.js [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+  }
+
+  h3 + p {
+    margin: 0;
+  }
+`
+
+const Objectives = ({ description }) => {
+  const descriptionSplitting = description
+    .replace(/- /g, '')
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Why?

File: src/client/TrainingPdf.js [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+
+const Picture = styled.img`
+  width: 50px;
+  float: left;
+`
+
+const AbstractParagraph = styled(ParagraphPrint)`
+  height: 50px;
+  width: 380px;
+  padding-left:60px;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: space

File: src/client/TrainingPdf.js [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+  margin-top:40px;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: space

File: src/client/TrainingPdf.js [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+  font-size:12px;
+  margin:0px;
+`
+
+const MainTitleSmall = styled(MainTitle)`
+  font-size: 20px;
+`
+
+const AbstractContent = styled.div`
+  height: 50px;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: line break before new selector please.

File: src/client/TrainingPdf.js [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+const FloatRigthContainer = styled.div`float: right;`
+
+const ColoredLogo = styled(Logo)`
+  color: ${theme.colors.primary};
+  width: 160px;
+`
+
+const ParagraphPrint = styled(Paragraph)`
+  font-size:12px;
+  margin:0px;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: space

File: src/client/TrainingPdf.js [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+
+const FloatRigthContainer = styled.div`float: right;`
+
+const ColoredLogo = styled(Logo)`
+  color: ${theme.colors.primary};
+  width: 160px;
+`
+
+const ParagraphPrint = styled(Paragraph)`
+  font-size:12px;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: space

File: src/client/TrainingPdf.js [hunk 13]
```diff
@@ -0,0 +1,289 @@
...
+  flex-direction: column;
+
+  @media (min-width: ${theme.medias.phablet}) {
+    flex-direction: row;
+  }
+`
+
+const Content = styled.div`
+  flex: 1;
+  margin: 20px 20px 20px;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: `margin: 20px`

File: src/client/TrainingPdf.js [hunk 14]
```diff
@@ -0,0 +1,289 @@
...
+  margin: 20px 0 0 0;
+  font-size: 14px;
+  font-weight: bold;
+  color: ${theme.colors.primary};
+`
+
+const Planning = styled.div`
+  flex: 1;
+  display: flex;
+  margin: 5px 0 10px 0;
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: `margin: 5px 0 10px;`

File: src/client/TrainingPdf.js [hunk 15]
```diff
@@ -0,0 +1,289 @@
...
+  line-height: 1em;
+`
+
+const ProgrammeArray = ({ programmeContent }) => {
+  const daySplitting = programmeContent.split(/### Jour \d+/)
+  daySplitting.shift()
+  const res = daySplitting.map((day, i) =>
+    <Planning key={i}>
+      <ProgrammeDay>
+        <ProgrammeDayTitle>
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: C'est dommage d'avoir à réécrire du texte déjà écrit, c'est moche.

File: src/client/TrainingPdf.js (hunk 2 above)
Comment (neoziro) [inline src/client/TrainingPdf.js]: Splitting is done two times, should be refactored.

File: src/client/TrainingPdf.js [hunk 16]
```diff
@@ -0,0 +1,289 @@
...
+                <ColoredLogo />
+                <ParagraphPrint>www.smooth-code.com</ParagraphPrint>
+              </FloatRigthContainer>
+              <MainTitleSmall itemProp="name">
+                {`Formation ${training.name}`}
+              </MainTitleSmall>
+              <AbstractContent>
+                <FloatRigthContainer>
+                  <ParagraphPrint>
+                    <strong>DURÉE: </strong>
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Durée: => `text-transform: uppercase;`

File: src/client/TrainingPdf.js [hunk 17]
```diff
@@ -0,0 +1,289 @@
...
+                    {`${training.duration} ${training.duration > 1
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Make it generic in a module.

File: src/client/TrainingPdf.js [hunk 18]
```diff
@@ -0,0 +1,289 @@
...
+                      ? 'jours'
+                      : 'jours'}`}
+                    <br />
+                    <strong>PRIX: </strong>
+                    {`${training.price}`}
+                  </ParagraphPrint>
+                </FloatRigthContainer>
+                <Picture
+                  alt="icone"
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: Alt not very relevant..

File: src/client/TrainingPdf.js [hunk 19]
```diff
@@ -0,0 +1,289 @@
...
+                  src={
+                    training &&
+                    clUrl(training.cloudinary_id, 'c_scale,w_50,h_50,dpr_2')
+                  }
+                />
+                <AbstractParagraph itemProp="description">
```
Comment (neoziro) [inline src/client/TrainingPdf.js]: itemProp nop.

File: src/client/TrainingPdf.js (hunk 7 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: .replace(/- /g, '') => List-style: none

File: src/client/TrainingPdf.js (hunk 6 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: Refacto

File: src/client/TrainingPdf.js (hunk 3 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: I accept any good idea.

File: src/client/TrainingPdf.js (hunk 15 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: Same

File: src/client/TrainingPdf.js (hunk 2 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: Take a look at the refacto

Comment (jsfez) [inline src/client/TrainingPdf.js]: done

File: src/client/TrainingPdf.js (hunk 16 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: Act first, understand after... I used style tag in JSX but i need to know why that's better.

File: src/client/TrainingPdf.js (hunk 17 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: I'll

File: src/client/TrainingPdf.js (hunk 18 above)
Comment (jsfez) [inline src/client/TrainingPdf.js]: icone => {`${training.name} icon`}

**Pull request merged**